include/swift/Frontend/InputFile.h

```diff
@@ -68,10 +73,22 @@ class InputFile {
     return filename.equals("<stdin>") ? "-" : filename;
   }
 
-  const std::string &outputFilename() const { return OutputFilename; }
+  StringRef outputFilename() const { return PSPs.OutputFilename; }
+
+  PrimarySpecificPaths getPrimarySpecificPaths() const { return PSPs; }
+
+  void setPrimarySpecificPaths(const PrimarySpecificPaths &PSPs) {
+    this->PSPs = PSPs;
+  }
 
-  void setOutputFilename(StringRef outputFilename) {
-    OutputFilename = outputFilename;
+  StringRef dependenciesFilePath() const {
+    return getPrimarySpecificPaths().SupplementaryOutputs.DependenciesFilePath;
+  }
+  StringRef loadedModuleTracePath() const {
+    return getPrimarySpecificPaths().SupplementaryOutputs.LoadedModuleTracePath;
+  }
+  StringRef TBDPath() const {
```

Comment:
Why are these three special?

Reply:
Added a comment: 
    // The next set of functions provides access to those primary-specific paths accessed directly from an InputFile.

Reply:
That's missing the "as opposed to through FrontendInputsAndOutputs" part, but even then I'm confused as to what's special about these three. Why do they get accessed through an InputFile rather than through FrontendInputsAndOutputs?

Reply:
In these three cases, the caller already has an InputFile and requires the corresponding output path. But perhaps what's bothering you is a lack of abstraction? The caller could pass the filename in to FrontendInputsAndOutputs to get the PrimarySpecificPaths, or I could add a function to FrontendInputsAndOutputs to get the PrimarySpecificPaths when given an InputFile.  The latter would hide the topology better. What's the itch you're trying to scratch here?

Reply:
I guess I just think it's weird that sometimes you can go from an InputFile directly to an output path and sometimes you have to go through the PrimarySpecificPaths. It might just be that these are the only three that come up where you *have* an InputFile instead of just a PrimarySpecificPaths, but it still seems inconsistent.

(My previous comment was wrong. I thought you could go from a *filename* to a supplementary output path via FrontendInputsAndOutputs, but that doesn't exist, and shouldn't.)